CRM/Contact/Form/Relationship.php

```diff
@@ -111,23 +112,54 @@ public function getDefaultEntity() {
 
   public function preProcess() {
     $this->_contactId = $this->get('contactId');
-
-    $this->_contactType = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_Contact', $this->_contactId, 'contact_type');
-
     $this->_relationshipId = $this->get('id');
 
-    $this->_rtype = CRM_Utils_Request::retrieve('rtype', 'String', $this);
+    $contact = \Civi\Api4\Contact::get(FALSE)
+      ->addSelect('display_name', 'contact_type')
+      ->addWhere('id', '=', $this->_contactId)
+      ->execute()
+      ->first();
+    $this->_contactType = $contact['contact_type'];
+    $this->_display_name_a = $contact['display_name'];
+    $this->assign('display_name_a', $this->_display_name_a);
 
+    $this->_rtype = CRM_Utils_Request::retrieve('rtype', 'String', $this);
```

Comment:
Why is this line still here?

Reply:
Because currently in the case that the form is creating a relationship it will still be set by the form submit